api/storage_service.cc

```diff
@@ -288,6 +323,56 @@ void set_storage_service(http_context& ctx, routes& r) {
         }));
     });
 
+    ss::toppartitions_generic.set(r, [&ctx] (std::unique_ptr<request> req) {
+        bool filters_provided = false;
+
+        std::unordered_set<std::tuple<sstring, sstring>, utils::tuple_hash> table_filters {};
+        if (req->query_parameters.contains("table_filters")) {
+            filters_provided = true;
+            auto filters = req->get_query_param("table_filters");
+            std::stringstream ss { filters };
+            std::string filter;
+            while (!filters.empty() && ss.good()) {
+                std::getline(ss, filter, ',');
+                table_filters.emplace(parse_fully_qualified_cf_name(filter));
+            }
+        }
+
+        std::unordered_set<sstring> keyspace_filters {};
+        if (req->query_parameters.contains("keyspace_filters")) {
+            filters_provided = true;
+            auto filters = req->get_query_param("keyspace_filters");
+            std::stringstream ss { filters };
+            std::string filter;
+            while (!filters.empty() && ss.good()) {
+                std::getline(ss, filter, ',');
+                keyspace_filters.emplace(std::move(filter));
+            }
+        }
+
+        // when the query is empty return immediately
+        if (filters_provided && table_filters.empty() && keyspace_filters.empty()) {
```

Comment:
how can that be? someone either table_filters was true or keyspace_filters was true or filter_pfovided was false

Reply:
I still not completely understand, they added at least one of the query parameters (keyspace or table), but they added it without a value?
I'm not even sure if we support it, did you check that?

Reply:
yes
For simplicity, let's pretend there is no `keyspace_filters` parameter, only the table ones. Now, one of the two situations can happen:
    1. The user doesn't use the `table_filters` option -> the option is not in the params
    2. The user provides the 'table_filters' option -> the option is in the params and contains a list of table names
    
In the first case, the query runs on all tables and keyspaces as this is the default
In the second case, the query runs only on the tables provided in the 'table_filters' option. A list can be empty, of course, and that means the query runs on a empty set - therefore, it gives empty results.

Reply:
I understand the code, my question was about the query parameter, it does not hurt to add the check like you did, I wasn't sure you can pass an empty parameter, so it looked redundant.